src/main/java/org/sasanlabs/service/vulnerability/fileupload/UnrestrictedFileUpload.java

```diff
@@ -10,31 +18,149 @@
 import org.sasanlabs.service.vulnerability.bean.GenericVulnerabilityResponseBean;
 import org.sasanlabs.vulnerability.types.VulnerabilitySubType;
 import org.sasanlabs.vulnerability.types.VulnerabilityType;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
 
-/** @author KSASAN preetkaran20@gmail.com */
+/**
+ * @author KSASAN preetkaran20@gmail.com Special Thanks
+ *     <p>https://bezkoder.com/spring-boot-file-upload/
+ *     <p>https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects
+ *     <p>https://www.youtube.com/watch?v=CmF9sEyKZNo
+ */
 @VulnerableAppRestController(
         descriptionLabel = "UNRESTRICTED_FILE_UPLOAD_VULNERABILITY",
         value = "UnrestrictedFileUpload",
         type = {VulnerabilityType.UNRESTRICTED_FILE_UPLOAD})
 public class UnrestrictedFileUpload {
+    private final Path root;
 
-    /** https://bezkoder.com/spring-boot-file-upload/ READ it */
+    private static final String STATIC_FILE_LOCATION = "upload";
+    private static final String BASE_PATH = "static";
+    private static final Random RANDOM = new Random(new Date().getTime());
+
+    public UnrestrictedFileUpload() throws IOException, URISyntaxException {
+        URI uploadDirectoryURI =
```

Comment:
Should we have file paths in `application.properties`
 and read from there ? 
Given `VulnerableAppRestController` is already a `@RestController`

Reply:
hmm makes sense. Sure will move this to application.properties

Reply:
I was thinking again about it, i am not sure whether we need to add directory to application.properties as i think this  might not change in future and in case if we think it changes or being used at multiple places we can either move to a common constant or move to application.properties. WDYT ?

Reply:
Hmm , Let's say someone want to host this in a cloud server and the feel like they need to test this with a huge file of their own and they mount a volume that holds the file , in that case it might be easier for them to just change the path to the file in their own overidden app.properties , else they might have to tinker with the code again.
What do you think

Reply:
great example however the purpose of this vulnerability level is that user is only able to upload images but he is able to do many things and as a vulnerability it might not has that functionality of testing huge files.

I completely agree with your point that say NFS is mounted to some folder and they can change that folder in application.properties but i think being a vulnerable app this might never be the case and this is the vulnerability if user is able to upload for very big sizes as we want to restrict it to just images.

Reply:
Hmm I see , if we think that case is not going to come up , I think it should be fine 👍